### PR TITLE
first version of solution

### PR DESCRIPTION
**Pull request by Sashk52**

### CODE REVIEW

File: src/main/java/com/dev/movietheatre/model/dto/UserResponseDto.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package com.dev.movietheatre.model.dto;
+
+public class UserResponseDto {
+    private Long id;
+    private String email;
+    private String password;
```
Comment (MykolaFatkullin) [inline src/main/java/com/dev/movietheatre/model/dto/UserResponseDto.java]: Screenshot_1

File: src/main/java/com/dev/movietheatre/service/impl/UserMapperImpl.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        return userResponseDto;
+    }
+
+    @Override
+    public User dtoToUser(UserRequestDto userRequestDto) {
+        User user = new User();
+        user.setEmail(userRequestDto.getEmail());
+        user.setPassword(userRequestDto.getPassword());
+        return user;
+    }
```
Comment (MykolaFatkullin) [inline src/main/java/com/dev/movietheatre/service/impl/UserMapperImpl.java]: Do you need this method?

File: src/main/java/com/dev/movietheatre/model/dto/UserRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package com.dev.movietheatre.model.dto;
+
+public class UserRequestDto {
```
Comment (MykolaFatkullin) [inline src/main/java/com/dev/movietheatre/model/dto/UserRequestDto.java:3]: Do you need this Dto?

**Review by MykolaFatkullin — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/movietheatre/model/dto/UserRequestDto.java (hunk 3 above)
Comment (Sashk52) [reply src/main/java/com/dev/movietheatre/model/dto/UserRequestDto.java:3]: Yes, in AuthenticationController , registration method.

File: src/main/java/com/dev/movietheatre/model/dto/UserResponseDto.java (hunk 1 above)
Comment (Sashk52) [reply src/main/java/com/dev/movietheatre/model/dto/UserResponseDto.java]: done

File: src/main/java/com/dev/movietheatre/service/impl/UserMapperImpl.java (hunk 2 above)
Comment (Sashk52) [reply src/main/java/com/dev/movietheatre/service/impl/UserMapperImpl.java]: No need right now, made for future. Actually agree with you. Deleted.

**Review by MykolaFatkullin — APPROVED** [review]

File: src/main/java/com/dev/movietheatre/controllers/OrderController.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/movietheatre/controllers/OrderController.java]: redundant empty line

File: src/main/java/com/dev/movietheatre/controllers/ShoppingCartController.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/movietheatre/controllers/ShoppingCartController.java]: same

File: src/main/java/com/dev/movietheatre/controllers/UserController.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import com.dev.movietheatre.service.UserService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/movietheatre/controllers/UserController.java]: same

File: src/main/java/com/dev/movietheatre/controllers/UserController.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    public UserController(UserService userService, UserMapper userMapper) {
+        this.userService = userService;
+        this.userMapper = userMapper;
+    }
+
+    @GetMapping("/by-email")
+    public UserResponseDto getbyEmail(@RequestParam String email) {
+        User user = userService.findByEmail(email).get();
+        return userMapper.mapUserToDto(user);
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/movietheatre/controllers/UserController.java]: same

File: src/main/java/com/dev/movietheatre/model/dto/OrderResponseDto.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
+package com.dev.movietheatre.model.dto;
+
+import java.util.List;
+
+public class OrderResponseDto {
+    private Long id;
+    private String date;
+    private String email;
+    private List<Long> ticketsIds;
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/movietheatre/model/dto/OrderResponseDto.java]:
```suggestion
    private List<Long> ticketIds;
```

**Review by KseniiaMakarova — APPROVED** [review]

File: src/main/java/com/dev/movietheatre/model/dto/OrderResponseDto.java (hunk 8 above)
Comment (Sashk52) [reply src/main/java/com/dev/movietheatre/model/dto/OrderResponseDto.java]: done

File: src/main/java/com/dev/movietheatre/controllers/UserController.java (hunk 7 above)
Comment (Sashk52) [reply src/main/java/com/dev/movietheatre/controllers/UserController.java]: done

File: src/main/java/com/dev/movietheatre/controllers/UserController.java (hunk 6 above)
Comment (Sashk52) [reply src/main/java/com/dev/movietheatre/controllers/UserController.java]: done

File: src/main/java/com/dev/movietheatre/controllers/ShoppingCartController.java (hunk 5 above)
Comment (Sashk52) [reply src/main/java/com/dev/movietheatre/controllers/ShoppingCartController.java]: done

File: src/main/java/com/dev/movietheatre/controllers/OrderController.java (hunk 4 above)
Comment (Sashk52) [reply src/main/java/com/dev/movietheatre/controllers/OrderController.java]: done

**Pull request merged**